pandas/core/frame.py

```diff
@@ -4239,6 +4240,14 @@ def _set_value(
                 self.loc[index, col] = value
             self._item_cache.pop(col, None)
 
+        except InvalidIndexError:
+            # GH48729: Seems like you are trying to assign a value to a
+            # row when only scalar options are permitted
+            raise InvalidIndexError(
```

Comment:
Could you `raise InvalidIndexError(...) from` the original exception?

Reply:
The original exception is: 
`slice(None, None, None)
`
which I dont think is pretty useful as it is still does not let the user know what causes the issue

Reply:
Would you really like to use that?

Reply:
It's mainly to keep the context of where the original exception was raised by using `from original_err`

Reply:
Hi, addressed the changes you requested.
Please let me know if this what you were referring to